libavpipe/src/avpipe_xc.c

```diff
@@ -4238,13 +4238,13 @@ avpipe_probe(
 
 avpipe_probe_end:
     if (decoder_ctx.format_context) {
-        if (decoder_ctx.format_context->pb->buffer){
-            AVIOContext *avioctx = (AVIOContext *) decoder_ctx.format_context->pb;
+        if (decoder_ctx.format_context->flags & AVFMT_FLAG_CUSTOM_IO) {
```

Comment:
This flag AVFMT_FLAG_CUSTOM_IO is only set in dashdec.c, be careful this logic doesn't leak memory again.

Can you run some valgrind and see if it leaks?

Reply:
I worried about it too but it's set reliably in avformat_open_input

```
int avformat_open_input(AVFormatContext **ps, const char *filename,
                        ff_const59 AVInputFormat *fmt, AVDictionary **options)

    ...
    if (s->pb) // must be before any goto fail
        s->flags |= AVFMT_FLAG_CUSTOM_IO;

```

Reply:
Yes, this is set in `avformat_open_input` as long as `pb` is set, which in our case is set at the end of `prepare_input`. I have not run it under valgrind, but am confident it is correct and _did_ check that it is set in relevant cases.

Reply:
Because if it leaks in line 4660, then it will leak badly.